### DISCUSSION TITLE
Risk-controlled selective gating for cross-session motor-imagery BCI

### DISCUSSION THREAD

**Discussion by xingsixue123** [PrismaX-Team/LiveDiscoveryBench-Tasks#19] [Proposals]
### Name

Sixue Xing

### Institution

University of Notre Dame

### Position

PhD Student

### Academic or professional profile

https://scholar.google.com/citations?user=xs6DLG4AAAAJ&hl=en

### Relevant experience

Motor-imagery BCI is part of my research. I am working on risk-aware decoding and on domain adaptation across sessions and across healthy and stroke subjects. My recent work brings conformal prediction and selective classification to motor imagery.

### Task title

Risk-controlled selective gating for cross-session motor-imagery BCI

### Scientific domain

brain-computer interfaces / neural engineering

### Scientific question

Given a frozen motor-imagery EEG classifier and a labelled session-1 calibration set, how should a system decide which trials to act on so that its error rate among those trials stays below a target r on a later session, while acting as often as possible?

This is risk-controlled selective classification under covariate shift. The classifier is fixed and cannot be retrained. Calibration comes from session 1(train source) and evaluation from session 2(test session, recorded on different day), so the exchangeability assumption that underpins finite-sample risk-control guarantees (conformal prediction, RCPS, Learn-then-Test) does not hold.

### Why it matters

There always exist a problem for all EEG interface: a decoder calibrated today works worse tomorrow, and how much worse varies unpredictably from person to person, which is "cross-session variability". This is why most systems still demand calibration before every use.

The general machine-learning answer is test-time adaptation: update at deployment using the unlabelled data that arrives during use. In a safety-relevant interface that is necessary but not sufficient. An adapted system cannot verify that it adapted correctly, because verification would need the labels it does not have; and a wrong action in a BCI is not a silent error, it is an action the user must notice and undo. What matters is therefore not average accuracy but a guarantee: whenever the system acts, it is right at least a stated fraction of the time.

Risk-controlled selective prediction is the framework that provides such a guarantee, and conformal prediction and its descendants claim to deliver it with finite-sample validity. Those guarantees rest on calibration and deployment data being exchangeable. Across EEG sessions they are not. Whether reliability guarantees can be made to survive that violation is open, and answering it would remove a concrete barrier to practical BCI use.

### Evidence it remains unresolved

Frontier decoders still cannot decode motor imagery reliably. Balanced accuracy on four-class BCI Competition IV-2a: EEGNet 0.4482, LaBraM-Base 0.4869, CBraMod 0.5138 [1]; REVE-Base, pretrained on 25,000 subjects and the strongest reported, reaches 0.6396 [2]. 

In our harness, frozen decoders sit in the same range: mean test accuracy on session2 is 0.608 over 315 (subject, model) pairs. Abstention is therefore the operating regime. At our operating point the best published gating method declines 55% of trials and is still only 71.0% accurate on the ones it accepts. Framed as risk control [3]: to hold the error rate at or below 0.30, the long-standing 70% BCI criterion, the best published gate can act on only 26.7% of trials, against 50.3% for an oracle that keeps the same confidence ordering and merely places the threshold correctly per subject. While at the 0.20 error rate target this task scores on, 15.3% against 32.2%. The gap is threshold placement under session shift.

[1] Wang et al., CBraMod: A Criss-Cross Brain Foundation Model for EEG Decoding, arXiv:2412.07236.
[2] El Ouahidi et al., REVE: A Foundation Model for EEG, arXiv:2510.21585.
[3] Geifman & El-Yaniv, Selective Classification for Deep Neural Networks, NeurIPS 2017.

### Current frontier

Coverage at target risk r = 0.20, macro-averaged. Per subject the score is the fraction of test session (session 2) trials acted on, or 0 if the error rate among them exceeds 0.20. Thresholds are set from train session (session-1) data only. Averaged within a cell, then unweighted over the 10 cells.

| Policy | bcic (4-class) | korea (2-class) | Macro |
| --- | --- | --- | --- |
| always-decide (no gate) | 0.0889 | 0.0778 | 0.0833 |
| Chow selective gate | 0.1362 | 0.0985 | 0.1174 |
| Softmax Response | 0.1658 | 0.0994 | 0.1326 |
| **THR conformal gate (best published)** | **0.1941** | **0.1114** | **0.1527** |
| *oracle threshold (not attainable)* | *0.3853* | *0.2583* | *0.3218* |

The strongest published method is the THR (vanilla threashold based conformal prediction) conformal gate at 0.1527, 1.83x the no-gate baseline. Conditions: train on session 1, calibrate on a held-out session-1 split, test on session 2; BCI Competition IV-2a (4 classes, 9 subjects, 216 test trials each) and OpenBMI (2 classes, 54 subjects, 150 test trials each); 5 backbones on each, 315 subject-model pairs.

### Measurable headroom

The SOTA to beat is **0.1527**, while the theoretical oracle threshold (same confidence ordering, threshold placed correctly for each subject) reaches **0.3218**. The gap is real but task is not easy, and progress should be expected in small steps rather than a jump. The missing quantity is how far each individual subject's session has drifted, and that is not recoverable from session-1 data by any straightforward route.

### Agent-visible inputs

Derived model outputs only — **no raw EEG**. The agent receives, for each of 10 episodes (5 backbones × 2 datasets):

| | bcic (BCI IV-2a) | korea (OpenBMI) |
| --- | --- | --- |
| classes | 4 | 2 |
| subjects | 9 | 54 |
| calibration trials / subject (session 1, **with labels**) | 72 | 50 |
| test trials / subject (session 2, **no labels**) | 216 | 150 |

Per subject: a calibration softmax matrix (N_cal × K) with its true labels, and a test softmax matrix (N_test × K) with the labels withheld. Plus episode metadata: number of classes and anonymised subject / backbone / dataset identifiers. In total 16,740 calibration rows and 50,220 test rows, about 2.8 MB.

All test rows for an episode are given **at once**, so transductive use of the unlabelled session-2 scores is allowed and is one of the intended attack surfaces. Session-2 trial order is shuffled per subject.

**No feedback is available during the task.** The agent calibrates on session 1 and is scored once on session 2, which mirrors real deployment: there is no oracle to query and no mid-run validation.

Every value was produced by a frozen, already-published inference pipeline.

### Expected submission

A gating **policy**, submitted as code, that drops in where `thr` / `chow` / `SR` sit. Scoring is fully automatic — the agent does not run the evaluation and never sees the session-2 labels.

The verifier calls `gate` once per subject-model pair — 315 calls — collects the decisions, and computes coverage at target risk from them.

**Requirements on the function**

- *Pure*: uses its arguments only. No file access, no network, no state carried between calls.
- *Deterministic*: identical arguments give identical output. The framework may repeat a call to check.
- *Bounded*: a few seconds per call is ample. Exceeding the budget, or raising, makes the submission invalid.
- Python standard library and numpy only.

A submission that abstains everywhere is valid but scores 0 (no coverage). One that acts everywhere is valid but violates the risk target on most subjects and also scores near 0.

### Environment and software dependencies

**Clean, no special dependencies.** The task is numeric post-processing of frozen softmax arrays; Python with numpy is sufficient. 

### Task-specific constraints

**Calibrate from session 1 only.** Session-2 labels are the hidden answer key. The unlabelled session-2 softmax may be used freely (transductive adaptation is in scope); the labels may not be obtained by any route.

**No lookup of the source datasets.** Do not use the internet to access, download or query BCI Competition IV-2a or OpenBMI (GigaDB 100542) label files, their mirrors, or any derived data containing the true labels of this task's test trials. Researching methods, papers and software online is allowed and expected.

Subject identities are anonymised and session-2 trial order is shuffled per subject, so released rows cannot be aligned to the public recordings by index. Network access and final artifacts are subject to audit.

No mid-run feedback.

### Estimated compute resources

**CPU only.** Roughly 4 cores, 8 GB RAM, ~1 hour of compute.

### Metric description

**Coverage at a target risk**, macro-averaged. Higher is better.

Per subject, with N session-2 trials, A trials acted on and W errors among those:

```
coverage        = A / N
selective error = W / A
score           = coverage   if selective error <= r
                = 0          otherwise
```

with **r = 0.20** — the system must be right at least 80% of the times it acts. Subject scores are averaged within an episode, then averaged unweighted across the 10 episodes.

This is the objective of risk-controlled selective classification: fix a target risk, maximise coverage (Geifman & El-Yaniv, NeurIPS 2017).

**Secondary metrics**, reported but not ranked on: the same quantity swept over r ∈ [0.10, 0.50]; F0.5; decision rate; decisive accuracy.

Invalid submissions — missing trials, malformed rows, labels outside the class set — are not ranked.

### Metric type

linear_maximize · Linear · higher is better 线性 · 越高越好

### References

BCI Competition IV-2a: `https://www.bbci.de/competition/iv/`, `https://pubmed.ncbi.nlm.nih.gov/22811657/`
OpenBMI: `https://doi.org/10.5524/100542`, `https://academic.oup.com/gigascience/article/8/5/giz002/5304369`

Actually, the task does not require the source data. What is shipped is derived model output only.

### Usage restrictions

Both source datasets may require citation: BCI Competition IV-2a, and OpenBMI.

### Permissions

- [X] I confirm that the data, code and other materials I provide have the permissions needed for task construction and evaluation in Science Innovation Exam, and that I have disclosed any attribution, access or redistribution restrictions.  
我确认所提供的数据、代码及其他材料具有用于 Science Innovation Exam 任务构建与评测的相应权限；如有署名、访问或再分发限制，我已说明。
- [X] If approved, I authorize Science Innovation Exam to use these materials to prepare a public task-repository contribution within the permissions and restrictions stated above.  
若提案获准，我授权 Science Innovation Exam 在上述权限和限制范围内，使用这些材料整理公开任务仓库贡献。
